token/services/ttx/multisig/wallet.go

```diff
@@ -115,20 +115,18 @@ func Wallet(sp token.ServiceProvider, wallet *token.OwnerWallet) *OwnerWallet {
 	}
 
 	tms := wallet.TMS()
-	nw := network.GetInstance(sp, tms.Network(), tms.Channel())
-	if nw == nil {
+	tokensProvider, err := tokens.GetProvider(sp)
+	if err != nil {
 		return nil
 	}
-	vault, err := nw.TokenVault(tms.Namespace())
+	tokens, err := tokensProvider.Tokens(tms.ID())
 	if err != nil {
-		logger.Errorf("failed to get vault for [%s:%s:%s]", tms.Network(), tms.Channel(), tms.Namespace())
 		return nil
 	}
-
 	return &OwnerWallet{
 		wallet:      wallet,
-		vault:       vault,
-		queryEngine: vault.QueryEngine(),
+		vault:       tokens,
+		queryEngine: tms.Vault().NewQueryEngine(),
```

Comment:
Do we need to have a unified query engine? Would it be better to split it into smaller engines? i.e. only for tokens.

Reply:
will open a Github issue

Reply:
https://github.com/hyperledger-labs/fabric-token-sdk/issues/989